internal/core/src/storage/Event.cpp

```diff
@@ -0,0 +1,357 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "storage/Event.h"
+#include "storage/Util.h"
+#include "storage/PayloadReader.h"
+#include "storage/PayloadWriter.h"
+#include "exceptions/EasyAssert.h"
+#include "utils/Json.h"
+#include "common/Consts.h"
+#include "common/FieldMeta.h"
+
+namespace milvus::storage {
+
+int
+GetFixPartSize(DescriptorEventData& data) {
+    return sizeof(data.fix_part.collection_id) + sizeof(data.fix_part.partition_id) + sizeof(data.fix_part.segment_id) +
+           sizeof(data.fix_part.field_id) + sizeof(data.fix_part.start_timestamp) +
+           sizeof(data.fix_part.end_timestamp) + sizeof(data.fix_part.data_type);
+}
+int
+GetFixPartSize(InsertEventData& data) {
+    return sizeof(data.start_timestamp) + sizeof(data.end_timestamp);
+}
+
+int
+GetEventHeaderSize(EventHeader& header) {
+    return sizeof(header.event_type_) + sizeof(header.timestamp_) + sizeof(header.event_length_) +
+           sizeof(header.next_position_);
+}
+
+int
+GetEventFixPartSize(EventType EventTypeCode) {
+    switch (EventTypeCode) {
+        case EventType::DescriptorEvent: {
+            DescriptorEventData data;
+            return GetFixPartSize(data);
+        }
+        case EventType::InsertEvent:
+        case EventType::DeleteEvent:
+        case EventType::CreateCollectionEvent:
+        case EventType::DropCollectionEvent:
+        case EventType::CreatePartitionEvent:
+        case EventType::DropPartitionEvent:
+        case EventType::IndexFileEvent: {
+            InsertEventData data;
+            return GetFixPartSize(data);
+        }
+        default:
+            PanicInfo("unsupported event type");
+    }
+}
+
+EventHeader::EventHeader(PayloadInputStream* input) {
+    auto ast = input->Read(sizeof(timestamp_), &timestamp_);
+    assert(ast.ok());
+    ast = input->Read(sizeof(event_type_), &event_type_);
+    assert(ast.ok());
+    ast = input->Read(sizeof(event_length_), &event_length_);
+    assert(ast.ok());
+    ast = input->Read(sizeof(next_position_), &next_position_);
+    assert(ast.ok());
+}
+
+std::vector<uint8_t>
```

Comment:
suggest to use "std::unique_ptr<uint8_t[]"
if return "std::vector<uint8_t>", then should "return std::move(res)"